### PR TITLE
SAA-736: Keep instances in sync with changes to activity start and end dates

### PR DESCRIPTION
**Pull request by gerryallan-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityService.kt [hunk 1]
```diff
@@ -315,7 +317,14 @@ class ActivityService(
   ) {
     request.startDate?.apply {
       activity.startDate = this
-      activity.schedules().forEach { it.startDate = this }
+      activity.schedules().forEach {
+        if (it.startDate < this) {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityService.kt:321]: I'd find it useful if would could put some of these conditions into functions with intention reveling names.

I am struggling a little to follow the rules.

Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityService.kt:321]: Yes, I agree - either function or a comment to describe how the from/to dates, when not null, affect the instances being created?

**Review by timharrison-moj — COMMENTED** [review]
LGTM, just a comment on clarifying the start/end date change actions?

**Review by matt-crowson-moj — APPROVED** [review]
LGTM - I still think we have some holes in the process but probably better to get this merged and discuss.

**Pull request merged**